features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java

```diff
@@ -113,6 +166,32 @@ public static ByteBuffer serialize(final Protocol protocol, final Iterable<Value
         return output.getByteBuffers().get(0).asNIO();
     }
 
+    protected void detectClockSkew(final long packetTimestamp, final InetAddress remoteAddress) {
+        if (getMaxClockSkew() > 0) {
+            long delta = Math.abs(packetTimestamp - System.currentTimeMillis());
+            if (delta > getMaxClockSkew() * 1000L) {
+                final Optional<Instant> instant = eventCache.getUnchecked(remoteAddress);
+
+                if (!instant.isPresent() || Duration.between(instant.get(), Instant.now()).getSeconds() > getClockSkewEventRate()) {
+                    eventCache.put(remoteAddress, Optional.of(Instant.now()));
+
+                    eventForwarder.sendNow(new EventBuilder()
+                            .setUei("uei.opennms.org/internal/telemetry/clockSkewDetected")
+                            .setTime(new Date())
+                            .setSource(getName())
+                            .setInterface(remoteAddress)
+                            .setDistPoller(identity.getId())
+                            .addParam("monitoringSystemId", identity.getId())
+                            .addParam("monitoringSystemLocation", identity.getLocation())
+                            .setParam("delta", (int) delta)
```

Comment:
Can we include the unit as part of the parameter name? i.e. `deltaMs`

Reply:
done